modules/handle_source.rb

```diff
@@ -0,0 +1,33 @@
+require_relative '../classes/source'
+require 'json'
+module HandleSource
+  def list_sources
+    puts 'No sources available' if @movies.empty?
+    puts '--------------------------'
+    puts 'List of Sources'
+    puts '--------------------------'
+    @sources.each_with_index do |source, index|
+      puts "#{index + 1} Name: #{source['name']}"
+    end
+  end
+
+  def save_new_source(source)
+    @sources = load_source
+    @sources << { id: source.id, name: source.name }
+    save_source(@sources)
+  end
+
+  def load_source
+    if File.exist?('./json/sources.json')
+      sources = File.open('./json/sources.json')
+      data = sources.read
+      JSON.parse(data)
```

Comment:
- [ ] A similar bug appears here so please follow the same approach to this one.

eg: 
```rb
      sources = (data.empty?) ? [] : JSON.parse(data)
```